lib/hiptest-publisher/options_parser.rb

```diff
@@ -203,14 +203,15 @@ def self.languages
     {
       'Ruby' => ['Rspec', 'MiniTest'],
       'Cucumber' => ['Ruby', 'Java', 'Javascript', 'Groovy', 'TypeScript'],
+      'Cucumber_5' => ['Java', 'Groovy'],
```

Comment:
That sounds fine to me!

Reply:
Thinking about this more, it feels a little awkward now.  It kind of feels like changing to `Cucumber_Expressions` would delegate the changes as being "this is done _specifically_ for Cucumber Expressions" and not "this is done to start bringing the features up to date."

For instance, the updated `actionwords.hbs` file uses the new imports,

```
import io.cucumber.datatable.DataTable;
import io.cucumber.java.en.*;
```

versus the old imports

```
import cucumber.api.DataTable;
import cucumber.api.java.en.*;
```

Which doesn't actually have anything to do with Expressions, and might result in somebody using the latest Java Cucumber exporting the old version instead of the new because they think they don't need Expressions.

I'm not sure what a good approach would be here off the top of my head, given the wide variance of versioning between the different Cucumber implementations.

Thoughts?

Reply:
Possibly: 'Cucumber_5+'

Not sure. Just a thought.

Reply:
> Possibly: 'Cucumber_5+'
> 
> Not sure. Just a thought.

Realistically (unfortunately?) that is still kind of a misnomer, given that Cucumber 5+ is not any real implication of a given set of features.  The cucumber-ruby gem, for instance, supports Expressions, and has since very `3.0.0.pre.2`.  Cucumber.js got it in 2.0.0.  The parity difference between versions is going to extend to more than just expressions, undoubtedly, which would make it all the more confusing.

Reply:
Yeah, I think you're right, that sounds awkward.

Maybe just renaming `Cucumber` as `Cucumber-legacy` for java, rename `Cucumber_5+` as `Cucumber`, and do a major release because this would break compatibility for people wanting to keep previous behavior when they use cucumber/java.

Thoughts?

Reply:
@cbliard 

That seems logical and straightforward.

So, just to recap, what I'll do is:

- Current `Cucumber` Java implementations (which would presumably be both Java and Groovy) become `Cucumber_Legacy`
- Current `Cucumber_5` Java implementations becomes `Cucumber`

A question to add to that, should the current `Cucumber` Typescript implementation be shifted to `Cucumber_Legacy` and the `Cucumber_Expressions` be shifted in with `Cucumber`?

Reply:
@cbliard I've gone ahead and addressed the 2 points above, updated the tests to reflect the changes, and setup the `Cucumber_Legacy` framework to act as an override to the `Cucumber` framework so that `Cucumber` will be the point of truth (that is, it will check for files in the `Cucumber_Legacy` folder first, then pull from the `Cucumber` folder, as opposed to the `Cucumber` framework pulling files from the `Cucumber_Legacy` folder)

I don't doubt that, as there continues to be a further deviation, some of the templates within the `Cucumber` folder will need to be copied over to the `Cucumber_Legacy` folder prior to modification, but no telling how far into the future that may be, and/or what files would require such action, so this felt like the best case scenario for now.

Awaiting your input regarding the `Cucumber_Expressions` framework.

Reply:
Awesome! Thanks a lot!

Yes, I think the same should be done for Typescript. Can you take care of it?

Reply:
@cbliard done, tests and readme updated as well!

Reply:
Excellent!

I'll try that manually and merge it. I hope to do it today.

Reply:
Looks good to me. I'll merge it.